pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -560,7 +553,7 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 		// Push only EDS. This is indexed already - push immediately
 		// (may need a throttle)
 		if len(con.Clusters) > 0 {
-			if err := s.pushEds(pushEv.push, con, pushEv.version, pushEv.edsUpdatedServices); err != nil {
+			if err := s.pushEds(pushEv.push, con, versionInfo(), pushEv.edsUpdatedServices); err != nil {
```

Comment:
why change the version, we need to use the version passed from xDSEvent

Reply:
I think `version ` is used to check if the pushing is out of date.  It is initiated in `func (s *DiscoveryServer) Push(full bool, edsUpdates map[string]struct{}) {
	if !full {`

And used here, pushConnection

```
	// check version, suppress if changed.
	currentVersion := versionInfo()
	if pushEv.version != currentVersion {
		adsLog.Infof("Suppress push for %s at %s, push with newer version %s in progress", con.ConID, pushEv.version, currentVersion)
		return nil
	}
```

and here `startPush`

```
		Retry:
			currentVersion := versionInfo()
			// Stop attempting to push
			if version != currentVersion && proxyFull {
				adsLog.Infof("PushAll abort %s, push with newer version %s in progress %v", version, currentVersion, time.Since(tstart))
				return
			}
```

Reply:
all of that code is gone now with this change. At least how it is now, I may need to make some more changes based on other comments

Reply:
Sorry, I can not understand what do u mean here?

Reply:
We no longer do 
```
			if version != currentVersion && proxyFull {
				adsLog.Infof("PushAll abort %s, push with newer version %s in progress %v", version, currentVersion, time.Since(tstart))
				return
			}
```

so we don't need the version anymore

Reply:
I doubt the version will change during one epoch push, for cds and eds, they can be different.